pinot-plugins/pinot-input-format/pinot-avro-base/src/main/java/org/apache/pinot/plugin/inputformat/avro/AvroUtils.java

```diff
@@ -353,16 +353,6 @@ public static Object handleMap(Map map) {
    * Handles the conversion of every field of the GenericRecord
    */
   private static Object handleGenericRecord(GenericData.Record record) {
-    List<Field> fields = record.getSchema().getFields();
-    if (fields.isEmpty()) {
-      return null;
-    }
-
-    Map<Object, Object> convertedMap = new HashMap<>();
-    for (Field field : fields) {
-      String fieldName = field.name();
-      convertedMap.put(fieldName, convert(record.get(fieldName)));
-    }
-    return convertedMap;
+    return record.get(0);
```

Comment:
I'm not sure if this is a valid fix. According to the comment instruction,

```
Converts the value to either a single value (string, number, bytebuffer), multi value (Object[]) or a Map
Natively Pinot only understands single values and multi values.
Map is useful only if some ingestion transform functions operates on it in the transformation layer.
```

Don't we need to follow the above conversion in case of `GenericData.Record`?

Reply:
This is following the conversion here. `GenericData.Record` is for fetching the value from a single value column.

Reply:
If GenericData.Record is supposed imply SV, then what is the distinction between the last two branches of this code

`public static Object convert(Object value) {
    if (value == null) {
      return null;
    }
    Object convertedValue;
    if (value instanceof Collection) {
      convertedValue = handleMultiValue((Collection) value);
    } else if (value instanceof Map) {
      convertedValue = handleMap((Map) value);
    } else if(value instanceof GenericData.Record) {
      convertedValue = handleGenericRecord((GenericData.Record) value);
    } else {
      convertedValue = RecordReaderUtils.convertSingleValue(value);
    }
    return convertedValue;
  }`

Reply:
Also, if the bug is the extraction of multi-value column, I don't quite follow how this fix is handling that. Shouldn't we be looking at handleMultiValue()?

Reply:
This is the original code to handle SV:
```
  public static Object handleSingleValue(@Nullable Object value) {
    if (value == null) {
      return null;
    }
    if (value instanceof GenericData.Record) {
      return handleSingleValue(((GenericData.Record) value).get(0));
    }
    return value;
  }
```
To determine whether it's SV or MV it's from the Pinot schema.

And the method `handleMultiValue()` would also call `convert()` method for every sub-value for MV column.